File: datasets_v2/tools/connectbot/swiftui/Act_generatepubkey.swift
import SwiftUI

struct Act_generatepubkey: View {
  var body: some View {
    ScrollView {
      VStack(alignment: .leading, spacing: 0) {
        HStack(alignment: .center, spacing: 0) {
          Text("\"Nickname:\"").multilineTextAlignment(.center)
          TextField("\"My work key\"", text: .constant("")).frame(maxWidth: .infinity)
        }
        HStack(alignment: .center, spacing: 0) {
          Text("\"Type:\"").multilineTextAlignment(.trailing).frame(maxHeight: .infinity, alignment: .center)
          VStack(alignment: .leading, spacing: 8) {
            /* TODO: RadioButton -> Toggle fallback */
            Toggle("RSA", isOn: .constant(false))
            /* TODO: RadioButton -> Toggle fallback */
            Toggle("DSA", isOn: .constant(false))
            /* TODO: RadioButton -> Toggle fallback */
            Toggle("EC", isOn: .constant(false))
            /* TODO: RadioButton -> Toggle fallback */
            Toggle("Ed25519", isOn: .constant(false))
          }
        }
        HStack(alignment: .center, spacing: 0) {
          Text("\"Bits:\"").multilineTextAlignment(.center)
          TextField("", text: .constant("")).frame(maxWidth: .infinity)
        }
        VStack(alignment: .leading, spacing: 0) {
        }
        .frame(maxWidth: .infinity)
        Text("\"Note: password can be blank\"").multilineTextAlignment(.leading)
        HStack(alignment: .center, spacing: 0) {
          Text("\"Password:\"").multilineTextAlignment(.center)
          TextField("", text: .constant("")).frame(maxWidth: .infinity)
        }
        HStack(alignment: .center, spacing: 0) {
          VStack(alignment: .trailing, spacing: 0) {
            Spacer()
            Text("\"Password:\"").multilineTextAlignment(.trailing)
            Text("\"(again)\"").multilineTextAlignment(.trailing)
            Spacer()
          }
          TextField("", text: .constant("")).frame(maxWidth: .infinity)
        }
        VStack(alignment: .leading, spacing: 0) {
        }
        VStack(alignment: .leading, spacing: 0) {
        }
        Button(action: { }) {
          Text("\"Generate\"")
        }
        .disabled(true)
      }
      .frame(maxWidth: .infinity)
    }
    .frame(maxWidth: .infinity)
  }
}

struct Act_generatepubkey_Previews: PreviewProvider {
  static var previews: some View {
    Act_generatepubkey()
  }
}

File: datasets_v2/tools/connectbot/compose/act_generatepubkey.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun act_generatepubkey() {
  Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().verticalScroll(rememberScrollState())) {
    Column(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
      Row() {
        Text(text = "\"Nickname:\"", textAlign = TextAlign.Center)
        OutlinedTextField(value = "", onValueChange = {}, placeholder = { Text("\"My work key\"") }, modifier = Modifier.wrapContentHeight().weight(1.0f))
      }
      Row() {
        Box(modifier = Modifier.fillMaxHeight(), contentAlignment = Alignment.CenterStart) {
        Text(text = "\"Type:\"", textAlign = TextAlign.End)
        }
        Column(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
          Row(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
            /* TODO: bind selection state */
            RadioButton(selected = false, onClick = {})
            Text("RSA")
          }
          Row(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
            /* TODO: bind selection state */
            RadioButton(selected = false, onClick = {})
            Text("DSA")
          }
          Row(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
            /* TODO: bind selection state */
            RadioButton(selected = false, onClick = {})
            Text("EC")
          }
          Row(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
            /* TODO: bind selection state */
            RadioButton(selected = false, onClick = {})
            Text("Ed25519")
          }
        }
      }
      Row() {
        Text(text = "\"Bits:\"", textAlign = TextAlign.Center)
        OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.wrapContentHeight().weight(1.0f))
      }
      Column(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
      }
      Text(text = "\"Note: password can be blank\"", modifier = Modifier.wrapContentWidth().wrapContentHeight(), textAlign = TextAlign.Start)
      Row() {
        Text(text = "\"Password:\"", modifier = Modifier.wrapContentWidth().wrapContentHeight(), textAlign = TextAlign.Center)
        OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.wrapContentWidth().wrapContentHeight().weight(1.0f))
      }
      Row() {
        Column(verticalArrangement = Arrangement.Center, horizontalAlignment = Alignment.End) {
          Text(text = "\"Password:\"", modifier = Modifier.wrapContentWidth().wrapContentHeight(), textAlign = TextAlign.End)
          Text(text = "\"(again)\"", modifier = Modifier.wrapContentWidth().wrapContentHeight(), textAlign = TextAlign.End)
        }
        OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.wrapContentWidth().wrapContentHeight().weight(1.0f))
      }
      Column(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
      }
      Column(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
      }
      Button(onClick = {}, enabled = false, modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
        Text(text = "\"Generate\"")
      }
    }
  }
}


@Preview(showBackground = true)
@Composable
fun act_generatepubkey_preview() {
    MyApplicationTheme {
        act_generatepubkey()
    }
}
